amethyst_animation/src/systems/control.rs

```diff
@@ -392,51 +321,55 @@ where
 
         // If pause was requested on a running animation, pause it
         (&ControlState::Running(..), &AnimationCommand::Pause) => {
-            pause_animation(control.id, hierarchy, samplers);
+            pause_animation(control.id, hierarchy, world);
             Some(ControlState::Paused(Duration::from_secs(0)))
         }
 
         // If start was requested on a paused animation, unpause it
         (&ControlState::Paused(_), &AnimationCommand::Start) => {
-            unpause_animation(control.id, hierarchy, samplers);
+            unpause_animation(control.id, hierarchy, world);
             Some(ControlState::Running(Duration::from_secs(0)))
         }
 
         (&ControlState::Running(..), &AnimationCommand::Step(ref dir)) => {
-            step_animation(control.id, hierarchy, samplers, sampler_storage, dir);
+            step_animation(control.id, hierarchy, world, sampler_storage, dir);
             None
         }
 
         (&ControlState::Running(..), &AnimationCommand::SetInputValue(value)) => {
-            set_animation_input(control.id, hierarchy, samplers, value);
+            set_animation_input(control.id, hierarchy, world, value);
             None
         }
 
         (&ControlState::Running(..), &AnimationCommand::SetBlendWeights(ref weights)) => {
-            set_blend_weights(control.id, hierarchy, samplers, weights);
+            set_blend_weights(control.id, hierarchy, world, weights);
             None
         }
 
         // check for finished/aborted animations, wait for samplers to signal done,
         // then remove control objects
         (&ControlState::Running(..), _) => {
-            if check_termination(control.id, hierarchy, &samplers) {
+            if check_termination(control.id, hierarchy, world) {
+                debug!("Animation finished, terminating.");
                 // Do termination
                 for node_entity in hierarchy.nodes.values() {
-                    let empty = samplers
-                        .get_mut(*node_entity)
+                    let empty = world
+                        .entry_mut(entity)
+                        .unwrap()
```

Comment:
I remember we had one time a overhaul to remove all unwraps. Either by replacing them by expects or something more sane. Should we replace this `unwrap()` with a `expect("Failed to resolve entity")` or something along the lines?

Reply:
i've replaced all in animation with expects